lib/experimental/media/class-gutenberg-rest-attachments-controller.php

```diff
@@ -0,0 +1,350 @@
+ */
+
+/**
+ * Class Gutenberg_REST_Attachments_Controller.
+ */
+class Gutenberg_REST_Attachments_Controller extends WP_REST_Attachments_Controller {
+	/**
+	 * Registers the routes for attachments.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes(): void {
+		parent::register_routes();
+
+		$args                       = $this->get_endpoint_args_for_item_schema();
+		$args['generate_sub_sizes'] = array(
+			'type'        => 'boolean',
+			'default'     => true,
+			'description' => __( 'Whether to generate image sub sizes.', 'gutenberg' ),
+		);
+		$args['convert_format']     = array(
+			'type'        => 'boolean',
+			'default'     => true,
+			'description' => __( 'Whether to support server-side format conversion.', 'gutenberg' ),
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $args,
+				),
+				'allow_batch' => $this->allow_batch,
+				'schema'      => array( $this, 'get_public_item_schema' ),
+			),
+			true
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\d]+)/sideload',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'sideload_item' ),
+					'permission_callback' => array( $this, 'sideload_item_permissions_check' ),
+					'args'                => array(
+						'id'         => array(
+							'description' => __( 'Unique identifier for the attachment.', 'gutenberg' ),
+							'type'        => 'integer',
+						),
+						'image_size' => array(
+							'description' => __( 'Image size.', 'gutenberg' ),
+							'type'        => 'string',
+							'required'    => true,
+						),
+					),
+				),
+				'allow_batch' => $this->allow_batch,
+				'schema'      => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Prepares a single attachment output for response.
+	 *
+	 * Ensures 'missing_image_sizes' is set for PDFs and not just images.
+	 *
+	 * @param WP_Post         $item    Attachment object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ): WP_REST_Response {
+		$fields   = $this->get_fields_for_response( $request );
+		$response = parent::prepare_item_for_response( $item, $request );
+
+		$data = $response->get_data();
+
+		if ( rest_is_field_included( 'missing_image_sizes', $fields ) ) {
```

Comment:
It looks like this is only here in order to additionally support PDF? Maybe worth clarifying in a comment, and also including a safeguard to only run this logic if the parent class hasn't already included it.
```suggestion
		$fields   = $this->get_fields_for_response( $request );
		$response = parent::prepare_item_for_response( $item, $request );

		$data = $response->get_data();

		// Handle missing image sizes for PDFs.
		if ( rest_is_field_included( 'missing_image_sizes', $fields ) && ! isset( $data['missing_image_sizes'] ) ) {
```